src/gui/update.rs

```diff
@@ -527,6 +529,104 @@ pub fn handle_message(ajour: &mut Ajour, message: Message) -> Result<Command<Mes
+                    ));
+                }
+            }
+
+            return Ok(Command::batch(commands));
+        }
+        Message::RepositoryPackagesFetched((flavor, result)) => {
+            match result.context(format!(
+                "Failed to fetch repository packages for {}",
+                flavor
+            )) {
+                Ok(packages) => {
+                    log::debug!(
+                        "Message::RepositoryPackagesFetched({}, {} packages)",
+                        flavor,
+                        packages.len()
+                    );
+
+                    let mut has_update = 0;
+
+                    let addons = ajour.addons.entry(flavor).or_default();
+                    let global_release_channel = ajour.config.addons.global_release_channel;
+
+                    // For each addon, check if an updated repository package exists. If it does,
+                    // we will apply that updated package to the addon, then check if
+                    // the addon is updatable.
+                    for addon in addons.iter_mut() {
+                        if let Some(package) = packages.iter().find(|p| {
+                            Some(p.id.as_str()) == addon.repository_id()
+                                && Some(p.kind) == addon.repository_kind()
+                        }) {
+                            // Update remote packages from refeshed repository package
+                            //
+                            // We don't want to replace the entire Repo Package of the addon
+                            // because we don't want to modify certain metadata such as File Id,
+                            // since we didn't use fingerprints to get these updated packages. We
+                            // just want to reference the "latest" remote packages from the repo,
+                            // and assign those to the Addon so we can check for new updates
+                            addon.set_remote_package_from_repo_package(package);
+
+                            // Check if addon is updatable.
+                            if let Some(package) =
+                                addon.relevant_release_package(global_release_channel)
+                            {
+                                if addon.is_updatable(&package) {
+                                    log::debug!(
+                                        "{} - Update is available for {}, {} -> {}",
+                                        flavor,
+                                        addon.title(),
+                                        addon.version().unwrap_or_default(),
+                                        package.version
+                                    );
+
+                                    addon.state = AddonState::Updatable;
+
+                                    has_update += 1;
+                                } else {
+                                    addon.state = AddonState::Idle;
```

Comment:
@casperstorm I'm thinking of removing this line. If an addon is updated, we show the status as "completed" until the user "refreshes". This would effectively overwrite those statuses every X minutes this process runs. What do you think?

Reply:
I agree, @tarkah. Sounds good.